app/main.py

```diff
@@ -1,25 +1,91 @@
 from abc import ABC
+from typing import Optional, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_value: int, max_value: int) -> None:
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = f"_{name}"
+
+    def __get__(
+            self,
+            instance: Optional["Visitor"],
+            owner: type
+    ) -> Optional[int]:
+        return getattr(instance, self.protected_name, None)
+
+    def __set__(self, instance: "Visitor", value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError
```

Comment:
The `TypeError` should include an informative message, e.g., `raise TypeError('Value must be an integer')`.